packages/block-editor/src/components/rich-text/index.native.js

```diff
@@ -589,7 +642,11 @@ export class RichText extends Component {
 			minHeight = style.minHeight;
 		}
 
-		const selection = this.state.needsSelectionUpdate ? { start: this.state.start, end: this.state.end } : null;
+		let selection = null;
+		if ( this.needsSelectionUpdate ) {
```

Comment:
Just curious, is there a specific rationale for moving the `needsSelectionUpdate` property from the `state` object to the `this` object?

Reply:
Hey @mkevins, yes, when I tested the code with needsSelectionUpdate being in the state of a component, I saw that we never reset it to false after it was being set to true and we only need needsSelectionUpdate to be true temporary, until we don't send start and end. 

When it was part of the state, if you want to reset it, you would need to make it be false, which means that you would force component to render one more time.

Reply:
That makes sense. Thank you for the explanation!